src/hash.c

```diff
@@ -84,6 +86,68 @@ FILE *hash_log;
 
 /* Hash algorithms */
 
+void MD5InitGeneric(void * mdContext) {
+    MD5Init((MD5_CTX *)mdContext);
+}
+
+void MD5UpdateGeneric(void * mdContext, const void * inBuf, size_t inLen) {
+    assert(inLen <= (size_t)UINT_MAX);
+    MD5Update((MD5_CTX *)mdContext, (const unsigned char *)inBuf, (unsigned int)inLen);
+}
+
+void MD5FinalGeneric(void * mdContext, void * buf) {
+    MD5Final((MD5_CTX *)mdContext, (char *)buf);
+}
+
+void SHA1InitGeneric(void * sc) {
+    SHA1Init((SHA1Context *)sc);
+}
+
+void SHA1UpdateGeneric(void * sc, const void * vdata, size_t len) {
+    assert(len <= (size_t)1u << 31);
```

Comment:
I'm a bit concerned about the asserts (and their absences in the other `xxxUpdateGeneric` functions). Is there any reason for those variations?

Reply:
@davidpolverari good to bring those two up, should be easy to explain:
- wrappers have an inner and an outer interface and for the "update" functions, the outer interface asks for `size_t` and the inner asks for something else, a different integer type
- when the target/inner integer type is (also unsigned in nature and) greater or equal in size, everything is fine, and no `assert` is needed.
- when it's smaller however, a wraparound/overflow/truncation would be possible and cause malfunction at runtime. Previous code did have that issue, it just kept silent about it, and now it because visible.
- My vote for first merging with the `assert` and then discuss ways to resolve them e.g. by changing the outer interface to `uint32_t` everywhere or by breaking the loop up and never feeding more than `UINT32_MAX` bytes and ones or something like that. The current `assert` is better than either uncontrolled behavior or hiding the issue to me.

What do you think?

Reply:
@davidpolverari PS: I fixed the value of one of the asserts.

Reply:
Thanks for the explanation. Now I see why you put asserts in some places and not in others and why you made different comparisons when asserting. Regarding their use, I see no problem using them at first. That said, I will have a final look at this whole PR later this evening when I arrive home before merging, but I think it is in good shape overall.

Reply:
@davidpolverari cool, thanks!